src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const options = Object.values(actions);
+  const result = [];
+
+  for (const option of options) {
+    const { keysToRemove, type, extraData } = option;
+    const stateCopy = { ...state };
+
+    switch (type) {
+      case 'addProperties':
+        let lastElementOfArray = null;
+
+        if (result[result.length - 1]) {
+          lastElementOfArray = Object.assign({}, result[result.length - 1],
+            extraData);
+        } else {
+          lastElementOfArray = Object.assign(stateCopy, extraData);
+        }
+        result.push(lastElementOfArray);
```

Comment:
why you need such a strange `if`? in such case you can just:
```suggestion
        Object.assign(stateCopy, extraData);
        result.push({ ...stateCopy })
```
didn't test it, but it should work